app/views/versions/index.html.erb

```diff
@@ -4,7 +4,7 @@
     <p><%= t('.not_hosted_notice') %></p>
   </div>
 <% else %>
-  <h3 class="t-list__heading"><%= t('.versions_since', :count => @versions.size, :since => nice_date_for(@rubygem.first_created_date)) %>:</h3>
+  <h3 class="t-list__heading"><%= t('.versions_since', :count => @versions.size, :since => nice_date_for(@versions.sort_by(&:created_at).first.created_at) %>:</h3>
```

Comment:
What about `@versions.map(&:created_at).min`?

Reply:
Nice! In my head creating 3000 strings was going to be slower than sorting the already-existing array, but looks like I was wrong.

```
Warming up --------------------------------------
              master     1.000  i/100ms
                  PR    11.000  i/100ms
               dkubb    61.000  i/100ms
Calculating -------------------------------------
              master      0.358  (± 0.0%) i/s -      2.000  in   5.587603s
                  PR    110.454  (± 4.5%) i/s -    561.000  in   5.088540s
               dkubb    612.666  (± 3.3%) i/s -      3.111k in   5.083449s

Comparison:
               dkubb:      612.7 i/s
                  PR:      110.5 i/s - 5.55x  slower
              master:        0.4 i/s - 1711.64x  slower
```

Reply:
I check the block format (`min { |a,b| }`) and it's also slower than the map/min method. Fascinating! Probably because symbol-to-proc is faster? IDK.

Reply:
I think the reason why it's faster is the difference between sorting and finding a minimum.

With a sort algorithm, you have to rank every item against every other item in the list based on some ranking algorith. To find the minimum is much simpler -- you just have to traverse the list one time, and store the current minimum value, comparing it against each entry and replacing it if you find something smaller.

Reply:
Another option to speed this up might be to ensure the `versions` are sorted by `created_at` when they are initially fetched from the DB. The association could be changed to scope `order`, eg:

```ruby
has_many :versions, -> { order(:created_at) }, dependent: :destroy, validate: false
```

It it could be scoped at the controller level

```ruby
Version.order(:created_at).scoping do
  # query that returns the Rubygem instance(s), assuming it uses includes to eager load the versions
end
```

If you *knew* the `versions` are already sorted then you can just do `@versions.first.created_at`.

Reply:
We can't sort the versions in the DB here because they have to be sorted by `:position` for the main view on this page.